### PR TITLE
[type: refactor] refactor netty configuration.

### PR DESCRIPTION
**Pull request by qicz**

### CODE REVIEW

File: shenyu-spring-boot-starter/shenyu-spring-boot-starter-gateway/src/main/java/org/apache/shenyu/springboot/starter/netty/ShenyuNettyWebServerConfiguration.java [hunk 1]
```diff
@@ -43,8 +43,8 @@
      */
     @Bean
     @ConfigurationProperties(prefix = "shenyu.netty.tcp")
-    public NettyTcpConfig nettyTcpConfig() {
-        return new NettyTcpConfig();
+    public NettyTcpConfigurationProperties nettyTcpConfigurationProperties() {
```
Comment (yu199195) [inline shenyu-spring-boot-starter/shenyu-spring-boot-starter-gateway/src/main/java/org/apache/shenyu/springboot/starter/netty/ShenyuNettyWebServerConfiguration.java:46]: will add conditionOnMissBean?

File: shenyu-spring-boot-starter/shenyu-spring-boot-starter-gateway/src/main/java/org/apache/shenyu/springboot/starter/netty/ShenyuNettyWebServerConfiguration.java [hunk 2]
```diff
@@ -54,42 +54,42 @@ public NettyTcpConfig nettyTcpConfig() {
      * @return the netty reactive web server factory
      */
     @Bean
-    public NettyReactiveWebServerFactory nettyReactiveWebServerFactory(final ObjectProvider<NettyTcpConfig> config) {
+    public NettyReactiveWebServerFactory nettyReactiveWebServerFactory(final ObjectProvider<NettyTcpConfigurationProperties> config) {
```
Comment (yu199195) [inline shenyu-spring-boot-starter/shenyu-spring-boot-starter-gateway/src/main/java/org/apache/shenyu/springboot/starter/netty/ShenyuNettyWebServerConfiguration.java:57]: add conditionOnMissBean?

File: shenyu-spring-boot-starter/shenyu-spring-boot-starter-gateway/src/main/java/org/apache/shenyu/springboot/starter/netty/ShenyuNettyWebServerConfiguration.java [hunk 3]
```diff
@@ -54,42 +54,42 @@ public NettyTcpConfig nettyTcpConfig() {
...
-        webServerFactory.addServerCustomizers(new EventLoopNettyCustomizer(nettyTcpConfig));
+        NettyTcpConfigurationProperties nettyTcpConfigurationProperties = Optional.ofNullable(config.getIfAvailable()).orElse(new NettyTcpConfigurationProperties());
+        webServerFactory.addServerCustomizers(new EventLoopNettyCustomizer(nettyTcpConfigurationProperties));
         return webServerFactory;
     }
 
     private static class EventLoopNettyCustomizer implements NettyServerCustomizer {
 
-        private final NettyTcpConfig nettyTcpConfig;
+        private final NettyTcpConfigurationProperties nettyTcpConfigurationProperties;
```
Comment (yu199195) [inline shenyu-spring-boot-starter/shenyu-spring-boot-starter-gateway/src/main/java/org/apache/shenyu/springboot/starter/netty/ShenyuNettyWebServerConfiguration.java:66]: nettyTcpConfigurationProperties  replac to  properties

File: shenyu-spring-boot-starter/shenyu-spring-boot-starter-gateway/src/main/java/org/apache/shenyu/springboot/starter/netty/ShenyuNettyWebServerConfiguration.java (hunk 1 above)
Comment (qicz) [reply shenyu-spring-boot-starter/shenyu-spring-boot-starter-gateway/src/main/java/org/apache/shenyu/springboot/starter/netty/ShenyuNettyWebServerConfiguration.java:46]: it is unecessary

File: shenyu-spring-boot-starter/shenyu-spring-boot-starter-gateway/src/main/java/org/apache/shenyu/springboot/starter/netty/ShenyuNettyWebServerConfiguration.java (hunk 2 above)
Comment (qicz) [reply shenyu-spring-boot-starter/shenyu-spring-boot-starter-gateway/src/main/java/org/apache/shenyu/springboot/starter/netty/ShenyuNettyWebServerConfiguration.java:57]: it is unnecessary

File: shenyu-spring-boot-starter/shenyu-spring-boot-starter-gateway/src/main/java/org/apache/shenyu/springboot/starter/netty/ShenyuNettyWebServerConfiguration.java (hunk 3 above)
Comment (qicz) [reply shenyu-spring-boot-starter/shenyu-spring-boot-starter-gateway/src/main/java/org/apache/shenyu/springboot/starter/netty/ShenyuNettyWebServerConfiguration.java]: rename to NettyTcpProperties. done!

**Comment by qicz** [comment]
the ci failure again , just rename the class.

**Review by yu199195 — APPROVED** [review]

**Pull request merged**